samples/EcsPipeline/main.cpp

```diff
@@ -0,0 +1,99 @@
+
+    auto scheduler =
+        Configurator()
+            .WithDefaultSystemsAndComponents()
+            .WithPluginProvider(selector.GetDefaultPluginTypeOnCurrentPlatformFor<IGraphicsPluginProvider>())
+            .WithPluginProvider(windowingProvider)
+            .WithPluginProvider(selector.GetDefaultPluginTypeOnCurrentPlatformFor<IResourceManagementPluginProvider>())
+            .InitialiseAndRegisterComponents();
+
+    std::shared_ptr<NovelRT::Ecs::Graphics::DefaultRenderingSystem> renderingSystem =
+        scheduler.GetRegisteredIEcsSystemAs<NovelRT::Ecs::Graphics::DefaultRenderingSystem>();
+
+    NovelRT::Experimental::Threading::FutureResult<NovelRT::Ecs::Graphics::TextureInfo> textureFuture =
+        renderingSystem->GetOrLoadTexture("novel-chan");
+
+    renderingSystem->ForceVertexTextureFutureResolution();
+
+    // unused(renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(),
+    // scheduler));
+
+    auto transformBuffer = scheduler.GetComponentCache().GetComponentBuffer<TransformComponent>();
+    auto entityGraphBuffer = scheduler.GetComponentCache().GetComponentBuffer<EntityGraphComponent>();
+
+    EntityId parentEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    EntityId childEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    EntityId childOfChildEntity =
+        renderingSystem->CreateSpriteEntityOutsideOfSystem(textureFuture.GetBackingConcurrentSharedPtr(), scheduler);
+
+    transformBuffer.PushComponentUpdateInstruction(0, childEntity, TransformComponent{ NovelRT::Maths::GeoVector3F(200, 200, 0), NovelRT::Maths::GeoVector2F::zero(), 0 });
+    transformBuffer.PushComponentUpdateInstruction(0, childOfChildEntity, TransformComponent{ NovelRT::Maths::GeoVector3F(200, 200, 0), NovelRT::Maths::GeoVector2F::zero(), 0 });
+    entityGraphBuffer.PushComponentUpdateInstruction(0, childEntity, EntityGraphComponent{true, parentEntity, 0});
+    entityGraphBuffer.PushComponentUpdateInstruction(0, childOfChildEntity, EntityGraphComponent{true, childEntity, 0});
+
+    scheduler.RegisterSystem(
+        [](auto delta, auto catalogue)
+        {
+            ComponentView<TransformComponent> transforms = catalogue.GetComponentView<TransformComponent>();
+
+            for (auto [entity, transform] : transforms)
+            {
+                TransformComponent newComponent{};
+                newComponent.rotationInEulerAngles = 20 * delta.getSecondsFloat();
+                newComponent.scale = NovelRT::Maths::GeoVector2F::zero();
+                transforms.PushComponentUpdateInstruction(entity, newComponent);
+            }
+        });
+
+    /*
+    std::random_device rd;                               // obtain a random number from hardware
```

Comment:
I'd also remove this code if it's in a sample and not being used.

Reply:
No idea what the final sample will look like yet for this. This is me just debugging different things at present.